src/lib/autoinstall/widgets/ask_dialog.rb

```diff
@@ -0,0 +1,434 @@
+
+        # Handles CWM events
+        #
+        # * timeout: updates the counter.
+        # * stop_counter or any "ValueChanged" event: stops the counter and
+        #   ignores any further timeout event.
+        #
+        # @param event [Hash] CWM event
+        def handle(event)
+          return if stopped?
+
+          if event["ID"] == :timeout
+            decrease_counter
+            return :next if timed_out?
+
+          elsif event["ID"] == :stop_counter || event["EventReason"] == "ValueChanged"
+            stop
+          end
+
+          nil
+        end
+
+        def contents
+          w = widgets.clone
+          w.concat([VStretch(), Label(Id(:counter), timeout.to_s)]) unless timeout.zero?
+          VBox(*w)
+        end
+
+        def cwm_definition
+          timeout.zero? ? super : super.merge("ui_timeout" => 1000)
+        end
+
+      private
+
+        # Decreases the time counter
+        def decrease_counter
+          @remaining -= 1
+          Yast::UI.ChangeWidget(Id(:counter), :Label, @remaining.to_s)
+        end
+
+        # Stops the counter
+        def stop
+          @stopped = true
+          Yast::UI.ChangeWidget(Id(:stop_counter), :Enabled, false)
+        end
+      end
+
+      # @macro seeDialog
+      attr_reader :disable_buttons
+
+      # Constructor
+      #
+      # @param dialog [Y2Autoinstall::Ask::Dialog] Dialog specification
+      # @param disable_back_button [Boolean] Whether the :back button should be disabled
+      def initialize(dialog, disable_back_button: false)
+        textdomain "autoinst"
+        @dialog = dialog
+        @disable_buttons = disable_back_button ? ["back_button"] : []
+      end
+
+      # @macro seeAbstractWidget
+      def title
+        dialog.title
+      end
+
+      # @macro seeAbstractWidget
+      def contents
+        in_frame = []
+        frametitle = nil
+
+        widgets = dialog.questions.each_with_object([]) do |question, all|
+          if question.frametitle != frametitle && !in_frame.empty?
+            all << Left(Frame(frametitle, VBox(*in_frame)))
+            all << VSpacing(1)
+            in_frame.clear
+          end
+
+          widget = Left(widget_for(question))
+          if question.frametitle
+            frametitle = question.frametitle
+            in_frame << widget
+          else
+            all << widget
+          end
+        end
+        widgets << Frame(frametitle, VBox(*in_frame)) unless in_frame.empty?
+
+        TimeoutWrapper.new(widgets, timeout: dialog.timeout)
+      end
+
+      # @macro seeDialog
+      def run
+        result = super
+        dialog.questions.each { |q| log.debug "#{q.text} -> #{q.value}" }
+        result
+      end
+
+    private
+
+      # Dialog specification
+      attr_reader :dialog
+
+      # Defines the dialog's layout
+      #
+      # It is redefined because the ButtonBox requires to have buttons with :ok and :cancel
+      # roles, and that's not the case in this dialog.
```

Comment:
maybe issue of CWM::Popup that can be fixed?

Reply:
It looks like it happens when using the `ButtonBox`. I am not sure whether we want to change it for all Pop-ups.